project/tests/test_firebase_model.py

```diff
@@ -0,0 +1,42 @@
+import unittest
+from project.models.firebase import FirebaseModel
+from project.db import db
+from project.app import app
+
+TEST_DB = 'test.db'
+
+app.config['DEBUG'] = True
+app.config['TESTING'] = True
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///test.db'
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = True
+
+db.init_app(app)
+
+
+class FirebaseModelTest(unittest.TestCase):
+    def setUp(self):
+        print("setUp")
+        # self.app = app.test_client()
+        db.drop_all()
+        db.create_all()
```

Comment:
I've been struggling to figure out what's wrong with these test suites.. Can't figure it out. Even dug up Passport project from old archives and tried to backport the setUp/createFixtures from there. To no avail..

At this point I would recommend dropping the stdlib unittest approach and use the `pytest` library I recommended some week back. Here's an example of how it's used in conjunction with a database in my old OAuth project: https://github.com/libris/xl_auth/pull/4/commits/234f3106534102f06e57c787b03bb41b177c8ed1

It's also the framework used in `qsck`, check this initial PR adding the basic structure ("starting small"): https://github.com/mblomdahl/qsck/pull/2

Another good idea from these two repos mentioned above is to keep tests in `./tests` package, and the application source in `./gateway` directory. The "`./project`" dir with nested tests dir is a bit unfortunate.

Good things to do:

- [ ] switch to pytest
- [ ] make the simplest ever pytest test that passes (one not requiring database to be setup)
- [ ] `./project -> ./gateway`
- [ ] `./project/tests -> ./tests`